From c018366f74aa14a2efbcac09c2468bd2e6681c2e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?B=C5=82a=C5=BCej=20Sowa?= <bsowa123@gmail.com>
Date: Mon, 19 Dec 2022 17:56:36 +0000
Subject: [PATCH 1/2] Fix superbuild install permission errors

---
 CMakeLists.txt         | 12 +++++++++++-
 cmake/SuperBuild.cmake |  2 +-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6706022..9b13356 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@ else()
 endif()
 
 find_package(ament_cmake REQUIRED)
-find_package(microxrcedds_agent REQUIRED)
+find_package(microxrcedds_agent REQUIRED PATHS ${PROJECT_BINARY_DIR}/temp_install)
 find_package(rosidl_cmake REQUIRED)
 find_package(fastcdr REQUIRED)
 find_package(fastrtps REQUIRED)
@@ -129,6 +129,16 @@ install(
     share/${PROJECT_NAME}
   )
 
+if(EXISTS ${PROJECT_BINARY_DIR}/temp_install/)
+  install(
+    DIRECTORY
+      ${PROJECT_BINARY_DIR}/temp_install/
+    DESTINATION
+      ${CMAKE_INSTALL_PREFIX}
+    USE_SOURCE_PERMISSIONS
+  )
+endif()
+
 if(UROSAGENT_GENERATE_PROFILE)
   set(_OUTPUT_PATH "${CMAKE_CURRENT_BINARY_DIR}/python")
 
diff --git a/cmake/SuperBuild.cmake b/cmake/SuperBuild.cmake
index bead827..a3a8aff 100644
--- a/cmake/SuperBuild.cmake
+++ b/cmake/SuperBuild.cmake
@@ -30,7 +30,7 @@ if(NOT xrceagent_FOUND)
             PREFIX
                 ${PROJECT_BINARY_DIR}/agent
             INSTALL_DIR
-                ${CMAKE_INSTALL_PREFIX}
+                ${PROJECT_BINARY_DIR}/temp_install
             CMAKE_CACHE_ARGS
                 -DCMAKE_CXX_COMPILER:FILEPATH=${CMAKE_CXX_COMPILER}
                 -DCMAKE_C_COMPILER:FILEPATH=${CMAKE_C_COMPILER}
-- 
2.34.1

